src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.operations.OperationHandler;
+import core.basesyntax.operations.OperationStrategy;
+import core.basesyntax.service.ShopService;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final Storage storage;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy, Storage storage) {
+        this.operationStrategy = operationStrategy;
+        this.storage = storage;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = operationStrategy.getHandler(transaction.getOperation());
+            handler.apply(transaction); // Це має бути правильна сигнатура

```

Comment:
Ensure that the `OperationHandler` returned from `operationStrategy.getHandler()` is not null before calling `apply()`. This will prevent `NullPointerException` if the strategy does not exist for a given operation.